packages/components/src/TextArea/TextArea.tsx

```diff
@@ -33,11 +33,9 @@ export const TextArea = ({
 }: TextAreaProps): JSX.Element => {
   const [internalValue, setInternalValue] = useState<
     string | number | readonly string[] | undefined
-  >(autogrow && !value ? defaultValue : undefined)
-  // ^ holds an internal state of the value so that autogrow can still work with uncontrolled textareas
-  // essentially forces the textarea into an (interally) controlled mode if autogrow is true and mode is uncontrolled
+  >(value ?? defaultValue ?? '')
```

Comment:
According to your comment below, `value` shouldn't exist right?
```suggestion
  >(defaultValue ?? '')
```

Reply:
We could get away with that yeah. It wouldn't matter for the actual behaviour, because value takes priority over this for autogrow, but it would create a situation where this internal value is not accurate when value is provided. I did this just for what I considered "correctness" but now that I think of it, there still could be a situation where this becomes incorrect if the consumer adjusts `value` without `onChange` running.

Reply:
I was thinking it would be ugly to only populate/update this state when needed, but it's probably worth it to save potential re-renders and what not so I've changed it now.